swarm/network/stream/messages.go

```diff
@@ -194,15 +197,17 @@ func (p *Peer) handleOfferedHashesMsg(req *OfferedHashesMsg) error {
 		return fmt.Errorf("error initiaising bitvector of length %v: %v", len(hashes)/HashSize, err)
 	}
 	wg := sync.WaitGroup{}
+	ctx, cancel := context.WithTimeout(context.Background(), syncBatchTimeout)
```

Comment:
no defer cancel?

Reply:
no, explicit cancel. but i will think it through, this part is likely buggy